### PR TITLE
Generate wav files from phrases

### PR DESCRIPTION
**Pull request by r-a303931**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -23,76 +23,162 @@ func readFileLines(path string) ([]string, error) {
...
 	tpv := datatypes.NewTaskProcessVariables(task, generator, tempDir)
 	defer func() {
 		results <- tpv.GetFinalLogs()
 	}()
 
 	if len(task.Description) > 0 {
-		tpv.Println("")
-		tpv.Println("*** ", tpv.Task.Description, " ***")
-		tpv.Println("")
+		fmt.Println("")
```
Comment (chrisvire) [inline main.go]: Why did you change this from `tpv.Println()` which includes it in the logs for the task?

File: main.go [hunk 2]
```diff
@@ -23,76 +23,162 @@ func readFileLines(path string) ([]string, error) {
...
+	fmt.Println("Phrase  : ", tpv.Task.PhraseFilename)
+	fmt.Println("Output  : ", tpv.Task.OutputFilename)
+	fmt.Println("Parameters : ", tpv.Parameters)
 
 	wavs := make(chan string)
 	go convertWav(wavs, tpv)
 
-	phrases := make(chan PhraseReadResults)
-	go readPhrasesFromFile(tpv.Task.PhraseFilename, phrases)
+	parsedPhrases := readPhrasesFromFile(tpv.Task.PhraseFilename)
```
Comment (chrisvire) [inline main.go]: No goroutines?

Couldn't you take the output channel from the readPhrasesFromFile as an input channel into the generateWavFilesFromPhrases?

**Review by chrisvire — COMMENTED** [review]
Waiting for comments...

File: main.go (hunk 1 above)
Comment (r-a303931) [reply main.go]: That was for debugging goroutines deadlocking. I'll redo the commit to go back to tpv.Println

File: main.go (hunk 2 above)
Comment (r-a303931) [reply main.go]: The readPhrasesFromFile creates the channel and spawns a goroutine internally, so that the management of the goroutine and channel can happen inside of the function, e.g. on lines 60-61

**Review by chrisvire — CHANGES_REQUESTED** [review]
This is not correctly pipelining processes together.

The output channel from one goroutine should be the input channel to the next (for the phrase processing).

**Review by chrisvire — APPROVED** [review]
I will accept for now.  It still doesn't seem correct.
